File: app/src/test/java/com/example/notphilphil/bob/controllers/TestLoginActivity.java
package com.example.notphilphil.bob.controllers;


import android.test.mock.MockContext;

import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;

import java.io.File;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.OutputStreamWriter;

import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertTrue;

public class TestLoginActivity {
    static File file;

    @BeforeAll
    public static void buildFile() {
        LoggedUser.setTesting(true);
        file = new File("regUsers.txt");
    }

    private void addUser(String name, String email, String pass, LoggedUser.PermissionsEnum usertype) {
        try {
            OutputStreamWriter outWriter =
                    new OutputStreamWriter(new FileOutputStream(file, true));
            String toWrite = name+","+email+","+pass+","+usertype+"\n";
            outWriter.write(toWrite);
            outWriter.close();
        } catch (IOException err) {

        }
    }

    @Test
    public void testLogin() {
        LoginActivity testClass = new LoginActivity();
        String user = "phil3";
        String pass = "lip3";
        assertFalse(testClass.loginPressed(user, pass, file, new MockContext()));
    }

    @Test
    public void testLoginUserExists() {
        LoginActivity testClass = new LoginActivity();
        String user = "phil";
        String email = "pass";
        String pass = "lip";
        LoggedUser.PermissionsEnum user_type = LoggedUser.PermissionsEnum.USER;
        addUser(user, email, pass, user_type);
        assertTrue(testClass.loginPressed(user, pass, file, new MockContext()));
    }
}

File: app/src/test/java/com/example/notphilphil/bob/controllers/TestLoginActivity.kt
package com.example.notphilphil.bob.controllers


import android.test.mock.MockContext

import org.junit.jupiter.api.BeforeAll
import org.junit.jupiter.api.Test

import java.io.File
import java.io.FileOutputStream
import java.io.IOException
import java.io.OutputStreamWriter

import org.junit.jupiter.api.Assertions.assertFalse
import org.junit.jupiter.api.Assertions.assertTrue

internal class TestLoginActivity {

    private fun addUser(name: String, email: String, pass: String, usertype: LoggedUser.PermissionsEnum) {
        try {
            val outWriter = OutputStreamWriter(FileOutputStream(file!!, true))
            val toWrite = "$name,$email,$pass,$usertype\n"
            outWriter.write(toWrite)
            outWriter.close()
        } catch (err: IOException) {

        }

    }

    @Test
    fun testLogin() {
        val testClass = LoginActivity()
        val user = "phil3"
        val pass = "lip3"
        assertFalse(testClass.loginPressed(user, pass, file!!, MockContext()))
    }

    @Test
    fun testLoginUserExists() {
        val testClass = LoginActivity()
        val user = "phil"
        val email = "pass"
        val pass = "lip"
        val user_type = LoggedUser.PermissionsEnum.USER
        addUser(user, email, pass, user_type)
        assertTrue(testClass.loginPressed(user, pass, file!!, MockContext()))
    }

    companion object {
        private var file: File? = null

        @BeforeAll
        fun buildFile() {
            //LoggedUser.setTesting(true);
            file = File("regUsers.txt")
        }
    }
}
